dvc/output.py

```diff
@@ -368,7 +368,11 @@ def __init__(
             name=self.hash_name,
             value=getattr(self.meta, self.hash_name, None),
         )
-        if self.meta.nfiles or self.hash_info and self.hash_info.isdir:
+        if self.files:
+            tree = Tree.from_list(self.files, hash_name=self.hash_name)
+            tree.digest()
+            self.hash_info = tree.hash_info
```

Comment:
This introduces overhead but need to check how big it is

Reply:
Also, `self.meta` is being lost here, before it was restored from the `.dvc` file. 
Is still unclear to me how `meta` is used and if it's needed at all in this scenario